sound/soc/sof/ipc4-loader.c

```diff
@@ -423,6 +423,65 @@ int sof_ipc4_reload_fw_libraries(struct snd_sof_dev *sdev)
 	return ret;
 }
 
+/**
+ * sof_ipc4_update_cpc_from_manifest - Update the cpc in base config from manifest
+ * @sdev: SOF device
+ * @fw_module: pointer struct sof_ipc4_fw_module to parse
+ * @basecfg: Pointer to the base_config to update
+ */
+void sof_ipc4_update_cpc_from_manifest(struct snd_sof_dev *sdev,
+				       struct sof_ipc4_fw_module *fw_module,
+				       struct sof_ipc4_base_module_cfg *basecfg)
+{
+	const struct sof_man4_module_config *fw_mod_cfg;
+	u32 cpc_pick = 0;
+	u32 max_cpc = 0;
+	int i;
+
+	if (!fw_module || !fw_module->fw_mod_cfg)
+		return;
```

Comment:
is this not an error?

It'd be worth a comment to explain when this branch is taken and why

Reply:
This should not happen, I have it here for safety.
We also have it in other places.